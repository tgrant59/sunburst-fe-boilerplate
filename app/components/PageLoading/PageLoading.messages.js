export default {
    loadingMessage: 'Steeping Tea...',
    errorMessage: 'Error Loading Component',
}
