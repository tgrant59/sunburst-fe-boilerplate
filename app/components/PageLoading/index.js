import React from 'react'
import PropTypes from 'prop-types'

import messages from './PageLoading.messages'

const PageLoading = ({ error, pastDelay }) => {
    if (error) {
        return <p>{messages.errorMessage}</p>
    } else if (pastDelay) {
        return <p>{messages.loadingMessage}</p>
    }
    return null
}

PageLoading.propTypes = {
    error: PropTypes.bool,
    pastDelay: PropTypes.bool,
}

PageLoading.defaultProps = {
    error: false,
    pastDelay: false,
}

export default PageLoading
