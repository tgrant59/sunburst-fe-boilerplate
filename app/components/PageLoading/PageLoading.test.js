import React from 'react'
import { shallow } from 'enzyme'
import PageLoading from './index'

describe('<PageLoading />', () => {
    it('should match snapshot', () => {
        const renderedComponent = shallow(<PageLoading />)
        expect(renderedComponent).toMatchSnapshot()
    })

    it('should match snapshot when there is an error', () => {
        const renderedComponent = shallow(<PageLoading error />)
        expect(renderedComponent).toMatchSnapshot()
    })

    it('should match snapshot when past delay', () => {
        const renderedComponent = shallow(<PageLoading pastDelay />)
        expect(renderedComponent).toMatchSnapshot()
    })
})
