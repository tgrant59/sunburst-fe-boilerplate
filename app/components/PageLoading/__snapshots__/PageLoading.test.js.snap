// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageLoading /> should match snapshot 1`] = `""`;

exports[`<PageLoading /> should match snapshot when past delay 1`] = `
<p>
  Steeping Tea...
</p>
`;

exports[`<PageLoading /> should match snapshot when there is an error 1`] = `
<p>
  Error Loading Component
</p>
`;
